File: LeetCode/Simulation/JS/Leetcode2027-minimum-moves-to-convert-string.js
var minimumMoves = function(s) {
    let [ans, n] = [0, s.length];
    for(let i = 0;i < n;i++) {
        if (s[i] == 'O') continue;
        i += 2;
        ans++;
    }
    return ans;
};

File: LeetCode/Simulation/Rust/Leetcode2027-minimum-moves-to-convert-string.rs
// rust simulation, greedy。遇到X就+3，ans += 1
impl Solution {
    pub fn minimum_moves(s: String) -> i32 {
        let (mut ans, n) = (0, s.len());
        let mut i = 0;
        while i < n {
            if s.as_bytes()[i] == b'O' {
                i += 1;
                continue;
            }
            i += 3;
            ans += 1;
        }
        ans
    }
}
